File: Flutter-Learning-English-Application/lib/models/topic.dart
class Topic {
  final String id;
  final String topicName;
  final bool isPublic;
  final String owner;
  final int total;
  final String createAt;

  Topic({required this.id, required this.topicName, required this.isPublic, required this.owner, required this.total, required this.createAt});

  factory Topic.fromJson(Map<String, dynamic> json) {
    return Topic(
      id: json['_id'],
      topicName: json['topicName'],
      isPublic: json['isPublic'],
      owner: json['owner'],
      total: json['total'],
      createAt: json['createAt'],
    );
  }
}

File: Flutter-Learning-English-Server/models/Topic.js
const mongoose = require('mongoose')
const Schema = mongoose.Schema
const moment = require('moment')

const TopicSchema = new Schema({
    topicName: {
        type: String,
        require: true
    },
    isPublic: {
        type: Boolean,
        default: false
    },
    owner: {
        type: String,
        require: true
    }, // username in Account
    total: {
        type: Number,
        default: 0
    },
    createAt: {
        type: String,
        default: moment(Date.now()).format('YYYY-MM-DDTHH-mm-ss')
    }
})

module.exports = mongoose.model('Topic', TopicSchema)